src/core/algorithms/dd/split/split.cpp

```diff
@@ -353,51 +375,54 @@ std::vector<DF> Split::SearchSpace(model::ColumnIndex index) {
     std::size_t dif_num_rows = difference_typed_relation_->GetNumRows();
 
     model::TypedColumnData const& dif_column = difference_typed_relation_->GetColumnData(index);
-    model::Type const& type = dif_column.GetType();
 
     auto pair_compare = [](model::DFConstraint const& first_pair,
                            model::DFConstraint const& second_pair) {
         double const first_pair_length = first_pair.upper_bound - first_pair.lower_bound;
         double const second_pair_length = second_pair.upper_bound - second_pair.lower_bound;
         return (first_pair_length > second_pair_length) ||
-               (first_pair_length == second_pair_length &&
+               (model::IsEqual(first_pair_length, second_pair_length) &&
                 (first_pair.lower_bound > second_pair.lower_bound));
     };
 
     std::set<model::DFConstraint, decltype(pair_compare)> limits(pair_compare);
 
     // accepts a string in the following format: [a;b], where a and b are double type values
-    std::regex df_regex(R"(\[(\d{1,19}(\.\d*)?)\;(\d{1,19}(\.\d*)?)\]$)");
+    std::regex df_regex(R"(\[(.*)\;(.*)\]$)");
```

Comment:
We also could use `std::regex double_regex(R"([-+]?d*.?d+([eE][-+]?d+)?)");` to check that `a` and `b` are `double` values